// sakuranpm/src/index.js
const fetch = require('node-fetch');
let baseUrl = "https://api.skapi.ga/api/img";//Sakura Kinomoto Bot Api
module.exports = {
  /* Gifs */
  kiss: async function () {
    let api = await fetch(`${baseUrl}/kiss`).then(res => res.json());
    return api.url;
  },
  vomit: async function () {
    let api = await fetch(`${baseUrl}/vomit`).then(res => res.json());
    return api.url;
  },
  hug: async function () {
    let api = await fetch(`${baseUrl}/hug`).then(res => res.json());
    return api.url;
  },
  pat: async function () {
    let api = await fetch(`${baseUrl}/pat`).then(res => res.json());
    return api.url;
  },
  cry: async function () {
    let api = await fetch(`${baseUrl}/cry`).then(res => res.json());
    return api.url;
  },
  hi: async function () {
    let api = await fetch(`${baseUrl}/hi`).then(res => res.json());
    return api.url;
  },
  bye: async function () {
    let api = await fetch(`${baseUrl}/bye`).then(res => res.json());
    return api.url;
  },
  nope: async function () {
    let api = await fetch(`${baseUrl}/nope`).then(res => res.json());
    return api.url;
  },
  bored: async function () {
  	let api = await fetch(`${baseUrl}/bored`).then(res => res.json());
  	return api.url;
  },
  tickle: async function () {
    let api = await fetch(`${baseUrl}/tickle`).then(res => res.json());
    return api.url;
  },
  scared: async function () {
    let api = await fetch(`${baseUrl}/scared`).then(res => res.json());
    return api.url;
  }  


  /* Images */       
}
